src/server/cluster/cluster_family.cc

```diff
@@ -65,6 +65,43 @@ bool ClusterFamily::IsEnabledOrEmulated() const {
   return is_emulated_cluster_ || ClusterConfig::IsClusterEnabled();
 }
 
+// TODO: Extend this method to accommodate the needs of `CLUSTER NODES` and `CLUSTER SLOTS`.
+// TODO: Also make this function safe in that it will read the state atomically.
+ClusterConfig::ClusterShard ClusterFamily::GetEmulatedShardInfo(ConnectionContext* cntx) const {
+  ClusterConfig::ClusterShard info{
+      .slot_ranges = {{.start = 0, .end = ClusterConfig::kMaxSlotNum}},
+      .master = {},
+      .replicas = {},
+  };
+
+  ServerState& etl = *ServerState::tlocal();
+  if (etl.is_master) {
+    std::string cluster_announce_ip = absl::GetFlag(FLAGS_cluster_announce_ip);
+    std::string preferred_endpoint =
+        cluster_announce_ip.empty() ? cntx->owner()->LocalBindAddress() : cluster_announce_ip;
+
+    info.master = {.id = server_family_->master_id(),
+                   .ip = preferred_endpoint,
+                   .port = static_cast<uint16_t>(absl::GetFlag(FLAGS_port))};
+
+    auto vec = server_family_->GetDflyCmd()->GetReplicasRoleInfo();
+    if (!vec.empty()) {
+      info.replicas.push_back({.id = etl.remote_client_id_,
```

Comment:
why not for loop on vec?

Reply:
You're right. I copied from below. I'll also fix them later for the other commands.